master-firmware/tests/test_map.cpp

```diff
@@ -213,3 +213,11 @@ TEST(AMap, canSetCubesObstacleAtomically)
 
     map_set_cubes_obstacle(&map, 0, 0, 0, 0);
 }
+
+TEST(AMap, canSetTowerObstacleAtomically)
+{
+    mock().expectOneCall("chMtxLock").withPointerParameter("lock", &map.lock);
```

Comment:
Future idea: we could do something similar to std::lock_guard for chibios.

Reply:
using RAII?

Reply:
ja